geowrangler/datasets/utils.py

```diff
@@ -27,7 +28,7 @@ def urlretrieve(
             size = -1
             read = 0
             blocknum = 0
-            if "Content-length" in respheaders:
```

Comment:
What type respheaders? This should be caseless matching since headers are case insensitive.

Reply:
respheaders is HTTPMessage and confirming it is caseless. Will revert